### DISCUSSION TITLE
Strange Canister Shell crafting behaviour

### DISCUSSION THREAD

**Discussion by Kokhirus** [FakeFishGames/Barotrauma#14012] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Not sure, if it is a common problem, but this is not looking right. Maybe it is just a one time issue, so I will try to reproduce it later.
![image](https://github.com/FakeFishGames/Barotrauma/assets/91697233/5c78987b-f988-43a3-8d72-42cca78f8d5a)
UPD Have got a new strange behaviour while crafting Canister Shell (can be seen below). You can recraft Canister Shell using full Canister Shell for it:
![image](https://github.com/FakeFishGames/Barotrauma/assets/91697233/c99615f1-599e-46f2-bf84-460e500d4668)

### Reproduction steps

1. Approach fabricator to craft canister shell
2. Check resources
3. Get confused

### Bug prevalence

Just once

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.4.6.0 (Blood in the Water Update, hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Kokhirus** [comment]
Looks like the problem has something to do with canister shell's recipe via 3 steel bars. Game did allow me to craft a canister shell after I got an additional steel bar in fabricator:
![image](https://github.com/FakeFishGames/Barotrauma/assets/91697233/1df83233-4949-474f-9675-b39b6d065488)

**Comment by SomeRandomNoobKekeke** [comment]
> You can recraft Canister Shell using full Canister Shell for it

Yea, it's been like that since release

> Problem 1 (you deleted what happened btw)

It's probably caused by all those recipes combos, mb code for checking and taking resources is different and cause this mismatch
What if you take 1 copper and 2 steel and manually put them in fabricator, will it allow you to craft it?

**Reply by Kokhirus** [reply]
Actually, it does work this way, but you will have to have one more steel bar on you
![image](https://github.com/FakeFishGames/Barotrauma/assets/91697233/6cdcefab-013c-4acc-b745-db0b61cef896)
If you don't then, well
![image](https://github.com/FakeFishGames/Barotrauma/assets/91697233/35579f4e-0d00-4535-9efc-1a8f1b9baf0e)
Game checks only option with crafting shell via three steel bars in this case

**Comment by Regalis11** [comment]
Fixed in https://github.com/FakeFishGames/Barotrauma-development/commit/6c7000e4fa0067d3e72c3142ab11b3aee6bbf44a (commit in our private repo)

**Comment by Jasontti** [comment]
Tested the fix and found no issues.
Closing discussion as resolved.
